lib/md051.js

```diff
@@ -0,0 +1,46 @@
+// @ts-check
+
+"use strict";
+
+const { addError, forEachHeading, filterTokens } = require("../helpers");
+
+/**
+ * Converts a Markdown heading into an HTML fragment
+ * according to the rules used by GitHub.
+ *
+ * @param {string} string The string to convert.
+ * @returns {string} The converted string.
+ */
+function convertHeadingToHTMLFragment(string) {
+  return "#" + string
+    .toLowerCase()
+    .replace(/ /g, "-")
```

Comment:
I think instead of space, it needs to be all characters, except the ones you've already identified in the next line. That captures stuff like the emoji characters in the titles that are failing in some of the tests right now

Reply:
Maybe it is `\W` https://www.tutorialspoint.com/w-vs-w-in-javascript-regex

Reply:
The particular tests you're worried about seem to be passing already? Unless you mean that GitHub translates the links differently and the tests aren't catching that?

Reply:
I mean it seems the repo tests for Mocha are failing from the emoji characters in https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md
I'm not sure if there is a difference between GFM and CommomMark for those characters

Reply:
Actually, I might have it backwards. Clicking on the link on github gets https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md#bug-i-found-a-bug

Reply:
> The particular tests you're worried about seem to be passing already?

Yep, it should already work with Emojis, and there are already passing tests for that. :smile: